File: sdks/php/src/Types/Bot.php
<?php

declare(strict_types=1);

namespace Linktor\Types;

class Bot
{
    public string $id = '';
    public string $name = '';
    public ?string $description = null;
    public ?string $agentId = null;
    public ?string $flowId = null;
    public array $channelIds = [];
    public string $status = '';
    public ?BotConfig $config = null;
    public bool $enabled = true;
    public array $metadata = [];
    public ?\DateTimeImmutable $createdAt = null;
    public ?\DateTimeImmutable $updatedAt = null;

    public static function fromArray(array $data): self
    {
        $instance = new self();
        $instance->id = $data['id'] ?? '';
        $instance->name = $data['name'] ?? '';
        $instance->description = $data['description'] ?? null;
        $instance->agentId = $data['agentId'] ?? null;
        $instance->flowId = $data['flowId'] ?? null;
        $instance->channelIds = $data['channelIds'] ?? [];
        $instance->status = $data['status'] ?? '';
        $instance->config = isset($data['config']) ? BotConfig::fromArray($data['config']) : null;
        $instance->enabled = $data['enabled'] ?? true;
        $instance->metadata = $data['metadata'] ?? [];
        $instance->createdAt = isset($data['createdAt']) ? new \DateTimeImmutable($data['createdAt']) : null;
        $instance->updatedAt = isset($data['updatedAt']) ? new \DateTimeImmutable($data['updatedAt']) : null;
        return $instance;
    }
}

class BotConfig
{
    public ?string $welcomeMessage = null;
    public ?string $fallbackMessage = null;
    public int $handoffThreshold = 3;
    public bool $autoHandoff = true;
    public array $keywords = [];

    public static function fromArray(array $data): self
    {
        $instance = new self();
        $instance->welcomeMessage = $data['welcomeMessage'] ?? null;
        $instance->fallbackMessage = $data['fallbackMessage'] ?? null;
        $instance->handoffThreshold = $data['handoffThreshold'] ?? 3;
        $instance->autoHandoff = $data['autoHandoff'] ?? true;
        $instance->keywords = $data['keywords'] ?? [];
        return $instance;
    }
}

class BotStatus
{
    public string $id = '';
    public string $status = '';
    public bool $isRunning = false;
    public int $activeConversations = 0;
    public ?\DateTimeImmutable $startedAt = null;
    public ?string $errorMessage = null;

    public static function fromArray(array $data): self
    {
        $instance = new self();
        $instance->id = $data['id'] ?? '';
        $instance->status = $data['status'] ?? '';
        $instance->isRunning = $data['isRunning'] ?? false;
        $instance->activeConversations = $data['activeConversations'] ?? 0;
        $instance->startedAt = isset($data['startedAt']) ? new \DateTimeImmutable($data['startedAt']) : null;
        $instance->errorMessage = $data['errorMessage'] ?? null;
        return $instance;
    }
}

File: sdks/rust/src/types/bot.rs
use serde::{Deserialize, Serialize};
use std::collections::HashMap;

#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "lowercase")]
pub enum BotStatus {
    Active,
    Inactive,
    Draft,
}

#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "lowercase")]
pub enum BotType {
    Flow,
    Ai,
    Hybrid,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Bot {
    pub id: String,
    pub tenant_id: String,
    pub name: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub description: Option<String>,
    pub status: BotStatus,
    #[serde(rename = "type")]
    pub bot_type: BotType,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub config: Option<HashMap<String, serde_json::Value>>,
    #[serde(default)]
    pub channel_ids: Vec<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub flow_id: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub agent_id: Option<String>,
    #[serde(default)]
    pub knowledge_base_ids: Vec<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub metadata: Option<HashMap<String, serde_json::Value>>,
    pub created_at: chrono::DateTime<chrono::Utc>,
    pub updated_at: chrono::DateTime<chrono::Utc>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateBotInput {
    pub name: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub description: Option<String>,
    #[serde(rename = "type")]
    pub bot_type: BotType,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub config: Option<HashMap<String, serde_json::Value>>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub channel_ids: Option<Vec<String>>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub flow_id: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub agent_id: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub knowledge_base_ids: Option<Vec<String>>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub metadata: Option<HashMap<String, serde_json::Value>>,
}

impl CreateBotInput {
    pub fn new(name: impl Into<String>, bot_type: BotType) -> Self {
        Self {
            name: name.into(),
            description: None,
            bot_type,
            config: None,
            channel_ids: None,
            flow_id: None,
            agent_id: None,
            knowledge_base_ids: None,
            metadata: None,
        }
    }

    pub fn description(mut self, desc: impl Into<String>) -> Self {
        self.description = Some(desc.into());
        self
    }

    pub fn config(mut self, config: HashMap<String, serde_json::Value>) -> Self {
        self.config = Some(config);
        self
    }
}

#[derive(Debug, Clone, Default, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct UpdateBotInput {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub name: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub description: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub status: Option<BotStatus>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub config: Option<HashMap<String, serde_json::Value>>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub channel_ids: Option<Vec<String>>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub flow_id: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub agent_id: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub knowledge_base_ids: Option<Vec<String>>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub metadata: Option<HashMap<String, serde_json::Value>>,
}

#[derive(Debug, Clone, Default, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ListBotsParams {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub status: Option<BotStatus>,
    #[serde(rename = "type", skip_serializing_if = "Option::is_none")]
    pub bot_type: Option<BotType>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub channel_id: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub search: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub limit: Option<i32>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub page: Option<i32>,
}
